sstables/promoted_index_blocks_reader.hh

```diff
@@ -9,7 +9,7 @@
 #pragma once
 
 #include "consumer.hh"
-#include "types.hh"
+#include "types/types.hh"
```

Comment:
Are you sure this "types.hh" here wasn't supposed to include the types.hh in the current directory, i.e., sstables/types.hh?
But perhaps you corrected here an error: perhaps the intention was to really include types/types.hh, and it didn't do what the author thought it does because of this naming conflict, but it didn't matter because some other header file included here included the really needed type/stypes.hh? :-)

Reply:
actually, i was struggling on this. it seems the better way forward is to just remove it and to see if the source compile. so a couple of occurrences of `#include "types.h"` are removed in the latest revision.